editor/src/clj/internal/graph.clj

```diff
@@ -1158,20 +1148,21 @@
+                                                                                                      deps (cond-> (transient [])
+                                                                                                                   (and dep-labels (pos? (count dep-labels)))
+                                                                                                                   (as-> $ (reduce #(conj! %1 [node-id %2]) $ dep-labels)))
                                                                                                       ;; The closest overrides
-                                                                                                      deps (transduce (map #(pair % single-label)) conj! deps overrides)
+                                                                                                      deps (transduce (map #(pair % label)) conj! deps overrides)
                                                                                                       ;; The connected nodes and their outputs
-                                                                                                      deps (transduce (keep (fn [^Arc arc]
-                                                                                                                              (let [target-id (.target-id arc)
-                                                                                                                                    target-label (.target-label arc)]
-                                                                                                                                (when-let [dep-labels (get (input-deps basis target-id) target-label)]
-                                                                                                                                  [target-id dep-labels]))))
+                                                                                                      deps (transduce (comp
+                                                                                                                        (keep (fn [^Arc arc]
+                                                                                                                                (let [target-id (.target-id arc)
+                                                                                                                                      target-label (.target-label arc)]
+                                                                                                                                  (when-let [dep-labels (get (input-deps basis target-id) target-label)]
+                                                                                                                                    (mapv #(vector target-id %) dep-labels)))))
+                                                                                                                        cat)
```

Comment:
`keep` returning a `seq` followed by `cat` is just `mapcat`, isn't it? 😄

Reply:
And `mapcat` is a `map` followed by `cat`... I think mapcat is a bit cleaner though, I'll use it